include/highfive/H5File.hpp

```diff
@@ -72,11 +72,16 @@ class File : public Object,
  private:
     std::string _filename;
 };
+
 }  // namespace HighFive
 
+
+// H5File is the main user constructible -> bring in implementation headers
+#include "bits/H5Annotate_traits_misc.hpp"
 #include "bits/H5File_misc.hpp"
-#include "H5Attribute.hpp"  // for AnnotateTraits
-#include "H5Group.hpp"      // for Node
+#include "bits/H5Node_traits_misc.hpp"
+#include "bits/H5Object_misc.hpp"
```

Comment:
So I understand correctly that `H5Object_misc.hpp` is included here because it will be needed in user code?

Reply:
Apart from the fact that `H5File.hpp` was promoted to the main header, implementation of headers which were simplified must be brought where those things are used.
Now, you have a point here since H5Object, as you mentioned below, already bring bits.
I'll another pass of checking these cases